apps/api-documenter/src/documenters/YamlDocumenter.ts

```diff
@@ -677,11 +650,46 @@ export class YamlDocumenter {
 
   private _getYamlItemName(apiItem: ApiItem): string {
     if (apiItem.parent && apiItem.parent.kind === ApiItemKind.Namespace) {
-      // For members a namespace, show the full name excluding the package part:
-      // Example: excel.Excel.Binding --> Excel.Binding
-      return this._getUid(apiItem).replace(/^[^.]+\./, '');
+      // If the immediate parent is a namespace, then add the namespaces to the name.  For example:
+      //
+      //   // Name: "N1"
+      //   export namespace N1 {
+      //     // Name: "N1.N2"
+      //     export namespace N2 {
+      //       // Name: "N1.N2.f(x,y)"
+      //       export function f(x: string, y: string): string {
+      //         return x + y;
+      //       }
+      //
+      //
+      //       // Name: "N1.N2.C"
+      //       export class C {
+      //         // Name: "member(x,y)"  <===========
+      //         public member(x: string, y: string): string {
+      //           return x + y;
+      //         }
+      //       }
+      //     }
+      //   }
+      //
+      // In the above example, "member(x, y)" does not appear as "N1.N2.C.member(x,y)" because YamlDocumenter
+      // embeds this entry in the web page for "N1.N2.C", so the container is obvious.  Whereas "N1.N2.f(x,y)"
+      // needs to be qualified because the DocFX template doesn't make pages for namespaces.  Instead, they get
+      // flattened into the package's page.
```

Comment:
@rbuckton Does the DocFX template have support for namespaces yet?  If so, then we could eliminate this awkward workaround and make proper pages for them.

For an example of how this is currently getting rendered, take a look at:  https://docs.microsoft.com/en-us/javascript/api/office?view=excel-js-preview#office-initialize

Reply:
No, it does not. We would have to submit a PR to dotnet/docfx for this change.

Reply:
I've mentioned this before, but I have a workaround I use for https://github.com/esfx/esfx that involves changes to support namespaces properly:

- https://github.com/esfx/esfx/blob/bbc36650efa6a88c9d998b0dafb53e8421513898/scripts/docs/yamlDocumenter.js#L107-L152
- https://github.com/esfx/esfx/blob/bbc36650efa6a88c9d998b0dafb53e8421513898/docs-sources/templates/esfx/UniversalReference.common.js#L19-L23
- https://github.com/esfx/esfx/blob/bbc36650efa6a88c9d998b0dafb53e8421513898/docs-sources/templates/esfx/partials/uref/namespace.tmpl.partial#L1 (NOTE: I redesigned the template organization to make it easier to tinker with)

Example: https://esfx.js.org/esfx/api/equatable/comparable_namespace.html#equatable_Comparable_Namespace

Reply:
Have you considered opening a PR for DocFX itself? I think everyone who uses TypeScript namespaces would be greatly appreciative of this change.

Reply:
I planned to do so eventually, but not until my schedule has settled down.